File: server/apps/backend/app/routers/rental_contracts.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from apps.backend.app.auth.deps import require_adult
from apps.backend.app.database import get_db
from apps.backend.app.models.user import User
from apps.backend.app.schemas.rental_contract import (
    RentalContractCreate,
    RentalContractResponse,
    RentalContractSummary,
    RentalContractUpdate,
)
from apps.backend.app.services import rental_contract as rental_service

router = APIRouter(prefix="/rental-contracts", tags=["rental-contracts"])


@router.get("", response_model=list[RentalContractResponse])
def list_rental_contracts(
    role: str | None = Query(None),
    active_only: bool | None = Query(None),
    db: Session = Depends(get_db),
    user: User = Depends(require_adult),
):
    return rental_service.list_rental_contracts(db, user, role, active_only)


@router.get("/summary", response_model=RentalContractSummary)
def get_rental_summary(
    db: Session = Depends(get_db),
    user: User = Depends(require_adult),
):
    return rental_service.get_rental_summary(db, user)


@router.post("", response_model=RentalContractResponse, status_code=201)
def create_rental_contract(
    req: RentalContractCreate,
    db: Session = Depends(get_db),
    user: User = Depends(require_adult),
):
    return rental_service.create_rental_contract(db, user, req)


@router.get("/{contract_id}", response_model=RentalContractResponse)
def get_rental_contract(
    contract_id: str,
    db: Session = Depends(get_db),
    user: User = Depends(require_adult),
):
    return rental_service.get_rental_contract(db, user, contract_id)


@router.patch("/{contract_id}", response_model=RentalContractResponse)
def update_rental_contract(
    contract_id: str,
    req: RentalContractUpdate,
    db: Session = Depends(get_db),
    user: User = Depends(require_adult),
):
    return rental_service.update_rental_contract(db, user, contract_id, req)


@router.delete("/{contract_id}")
def delete_rental_contract(
    contract_id: str,
    db: Session = Depends(get_db),
    user: User = Depends(require_adult),
):
    rental_service.delete_rental_contract(db, user, contract_id)
    return {"detail": "已删除"}

File: frontend/apps/main/src/api/rentalContracts.ts
import http from './index'
import type { RentalContract, RentalRequestPayload, RentalSummary } from '@/types'

export function getRentalContracts(params?: { role?: 'landlord' | 'tenant'; active_only?: boolean }) {
  return http.get<RentalContract[]>('/rental-contracts', { params })
}

export function getRentalContract(id: string) {
  return http.get<RentalContract>(`/rental-contracts/${id}`)
}

export function createRentalContract(data: RentalRequestPayload) {
  return http.post<RentalContract>('/rental-contracts', data)
}

export function updateRentalContract(id: string, data: RentalRequestPayload) {
  return http.patch<RentalContract>(`/rental-contracts/${id}`, data)
}

export function deleteRentalContract(id: string) {
  return http.delete(`/rental-contracts/${id}`)
}

export function getRentalSummary() {
  return http.get<RentalSummary>('/rental-contracts/summary')
}
